docs/_source/docs/faq.rst

```diff
@@ -95,6 +95,52 @@ be found in these projects:
 - `Sage-Bionetworks Sceptre lambda github template`_
 - `Versant SAM Sceptre skeleton example`_
 
+Another approach is to use Sceptre `before_create` and `before_update` hooks to
+run either the SAM or CFN "package" command to generate the packaged version of
+the template.
+
+CloudFormation Example:
+
+```yaml
+template_path: generated/lambda-packaged.json
+stack_name: cfn-lambda-example
+hooks:
+  before_create:
```

Comment:
i always use launch command which does both create and update.  seems like you can just do this with just one `before_launch` hook ?

Reply:
No the problem with `before_launch` is it executes for every stack. So if you have 10 stacks it will run 10 times. I've found `before_create` and `before_update` is the only way to get it to run once even when you run `sceptre launch`.

I saw in the dev branch there's a `before_generate` which I'd be curious if it will run only once per stack.

Reply:
i'm not sure i understand.  if i run `launch --ignore-dependencies prod/my_stack.yaml` the hook will only execute once for the stack defined in my_stack.yaml

Reply:
Do you usually run with `sceptre launch --ignore-depencencies prod/my_stack.yaml` to deploy each one of your stacks?

Normally I run like this `sceptre launch -y prod` to deploy everything under prod. If you do this using the `before_launch` hook, that hook will execute x times where x is the number of stacks that exist in the prod environment.

It seems the `before_launch` hook runs against every stack and not just the one stack when you are deploying multiple stacks.

Reply:
Just retested and it seems like `before_launch` is working as you describe. I'm not sure why I was having problems with it earlier but it's working for me now too. I'll update the PR.